pkg/migration/controllers/migration.go

```diff
@@ -1398,6 +1441,90 @@ func (m *MigrationController) applyResources(
 				"Resource migrated successfully")
 		}
 	}
+
+	// recreate pvc objects
+	for _, obj := range pvcObjects {
+		var pvc v1.PersistentVolumeClaim
+		var err error
+		if err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), &pvc); err != nil {
+			m.updateResourceStatus(
+				migration,
+				obj,
+				stork_api.MigrationStatusFailed,
+				fmt.Sprintf("Error applying resource: %v", err))
+			continue
+		}
+		deleteStart := metav1.Now()
+		if err = adminClient.CoreV1().PersistentVolumeClaims(pvc.GetNamespace()).Delete(context.TODO(), pvc.Name, metav1.DeleteOptions{}); err != nil {
+			log.MigrationLog(migration).Errorf("Error deleting %v %v during migrate: %v", pvc.Kind, pvc.GetName(), err)
+		} else {
+			for i := 0; i < deletedMaxRetries; i++ {
+				obj, err := adminClient.CoreV1().PersistentVolumeClaims(pvc.GetNamespace()).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
+				if err != nil && errors.IsNotFound(err) {
+					break
+				}
+				createTime := obj.GetCreationTimestamp()
+				if deleteStart.Before(&createTime) {
+					logrus.Warnf("Object[%v] got re-created after deletion. So, Ignore wait. deleteStart time:[%v], create time:[%v]",
+						obj.GetName(), deleteStart, createTime)
+					break
```

Comment:
why not mark the migration failed here with this error message? 
Anyway the migration is going to fail once we try to create the PVC on line1476. At least we will update with the real reason why it failed

Reply:
Migration will be marked as PartialSuccess since its failed as application stage.  We will be adding error we receive from create api in status field for failed pvc
`Error applying resource: <err from create pvc api`